models/issue_comment.go

```diff
@@ -543,6 +551,37 @@ func (c *Comment) CodeCommentURL() string {
 	return fmt.Sprintf("%s/files#%s", c.Issue.HTMLURL(), c.HashTag())
 }
 
+// LoadPushCommits Load push commits
+func (c *Comment) LoadPushCommits() (err error) {
+	if c.Content == "" || c.Commits != nil || c.Type != CommentTypePullPush {
+		return nil
+	}
+
+	commitIDs := strings.Split(c.Content, ":")
+
+	if c.IsForcePush {
+		if len(commitIDs) != 2 {
```

Comment:
Because I only add old commitID and newCommitID in here, So if number is not right, there must be an error in the database record. see:
https://github.com/a1012112796/gitea/blob/e3481b06dc562b43af0ddfaa40617f482fed96fe/models/issue_comment.go#L1116-L1124

Reply:
Oh, I see... the old/new commit id pair comes from way up the call tree. 👍